### PR TITLE
Hash.FromAnonymousObject also gets properties from base class.

### PR DESCRIPTION
**Pull request by farukterzioglu**
Fixes issue  #221

### CODE REVIEW

**Comment by laedit** [comment]
Thanks for the update 😄 

Can you just add some tests?

**Comment by farukterzioglu** [comment]
I wanted to add some test but couldn't find a proper place for testing this. (and delayed tests for later :( )
Is there a test class for Hash or do  I need to create a new test class for Hash behaviors? 

Can you lead me to the right way?

**Comment by laedit** [comment]
Sorry for not providing enough context.
Each source class should have the equivalent test class, in this case it seems that you have to create a new `HashTests` class.

File: src/DotLiquid/Hash.cs [hunk 1]
```diff
@@ -48,7 +48,9 @@ private static void FromAnonymousObject(object anonymousObject, Hash hash, bool
 
         private static Action<object, Hash> GetObjToDictionaryMapper(Type type, bool includeBaseClassProperties)
         {
-            if (!mapperCache.TryGetValue(type, out Action<object, Hash> mapper))
+            var cacheKey = type.FullName + (includeBaseClassProperties ? "WithBaseProperties" : "WithoutBaseProperties");
```
Comment (farukterzioglu) [inline src/DotLiquid/Hash.cs]: to use same class with/without base properties at the same time, caching should consider including base properties.

File: src/DotLiquid/Hash.cs [hunk 2]
```diff
@@ -24,25 +24,33 @@ public class Hash : IDictionary<string, object>, IDictionary
...
-            Action<object, Hash> mapper = GetObjToDictionaryMapper(anonymousObject.GetType());
+            Action<object, Hash> mapper = GetObjToDictionaryMapper(anonymousObject.GetType(), includeBaseClassProperties);
             mapper.Invoke(anonymousObject, hash);                
         }
 
-        private static Action<object, Hash> GetObjToDictionaryMapper(Type type)
+        private static Action<object, Hash> GetObjToDictionaryMapper(Type type, bool includeBaseClassProperties)
         {
-            if (!mapperCache.TryGetValue(type, out Action<object, Hash> mapper))
+            var cacheKey = type.FullName + (includeBaseClassProperties ? "WithBaseProperties" : "WithoutBaseProperties");
```
Comment (laedit) [inline src/DotLiquid/Hash.cs]: Can you add a separator between the type name and the suffix?

Comment (farukterzioglu) [reply src/DotLiquid/Hash.cs]: Done

**Comment by laedit** [comment]
Thanks!

**Pull request merged**